### PR TITLE
test: e2e & setup-test-lsm script

### PR DESCRIPTION
**Pull request by cybertronprime**
## 1. Summary
Fixes # (issue)
 Created a RegisterZone test  
 Updated dependencies to github.com/strangelove-ventures/interchaintest/v7 in test/interchaintest
 Updated some minor dependencies

## 2.Type of change

 
Adding a register-zone test file to the Interchain test script

## 5. Checklist

- [ ] Does the Readme need to be updated?

### CODE REVIEW

**Comment by aljo242** [comment]
@rajrohit10 lint failing

**Review by ajansari95 — CHANGES_REQUESTED** [review]
there seems to be some conflicts

File: test/interchaintest/setup.go [hunk 1]
```diff
@@ -36,7 +40,7 @@ func createConfig() (ibc.ChainConfig, error) {
 	return ibc.ChainConfig{
 			Type:                "cosmos",
 			Name:                "quicksilver",
-			ChainID:             "quicksilver-2",
+			ChainID:             "quicksilverd",
```
Comment (ajansari95) [inline test/interchaintest/setup.go:43]: chain-id not in desired format

File: test/interchaintest/interchainstaking_test.go [hunk 2]
```diff
@@ -230,6 +230,6 @@ chains:
 	err = xcc.StartContainer(ctx)
 	require.NoError(t, err)
 
-	err = xcc.Running(ctx)
-	require.NoError(t, err)
+	// err = xcc.Running(ctx)
+	// require.NoError(t, err)
```
Comment (ajansari95) [inline test/interchaintest/interchainstaking_test.go:234]: should be reverted

File: test/interchaintest/interchainstaking_test.go [hunk 3]
```diff
@@ -195,8 +195,8 @@ chains:
 	err = icq.StartContainer(ctx)
 	require.NoError(t, err)
 
-	err = icq.Running(ctx)
-	require.NoError(t, err)
+	// err = icq.Running(ctx)
+	// require.NoError(t, err)
```
Comment (ajansari95) [inline test/interchaintest/interchainstaking_test.go:199]: should be reverted

File: test/interchaintest/e2e_test.go [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+	client, network := interchaintest.DockerSetup(t)
+
+	ctx := context.Background()
+
+	cf := interchaintest.NewBuiltinChainFactory(zaptest.NewLogger(t), []*interchaintest.ChainSpec{
+		{
+			ChainConfig: ibc.ChainConfig{
+				Type:           "cosmos",
+				Name:           "quicksilver",
+				ChainID:        "quicksilverd",
```
Comment (ajansari95) [inline test/interchaintest/e2e_test.go:55]: chain-id not in desired format

**Review by ajansari95 — CHANGES_REQUESTED** [review]

File: test/interchaintest/setup.go (hunk 1 above)
Comment (rajrohit10) [reply test/interchaintest/setup.go]: Corrected

File: test/interchaintest/interchainstaking_test.go (hunk 3 above)
Comment (rajrohit10) [reply test/interchaintest/interchainstaking_test.go]: Will add in the next commit

**Comment by anhductn2001** [comment]
Hi @rajrohit10, I just checkout your branch and run ictest-registerzone. Now, the test is stuck due to len(connection) = 2. Can you check it?

**Comment by anhductn2001** [comment]
It seems this PR has an E2E part that overlaps with my previous PR #521 . So we will consider keeping the new changes and removing the duplicates.

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Pull request closed without merging**